packages/grid/_modules_/grid/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridSelector } from '../core/useGridSelector';
+import { gridContainerSizesSelector } from '../../../components/GridViewport';
+import { useNativeEventListener } from '../../root/useNativeEventListener';
+import { GRID_SCROLL } from '../../../constants/eventsConstants';
+
+export const useGridInfiniteLoader = (
+  windowRef: React.MutableRefObject<HTMLDivElement | null>,
+  apiRef: GridApiRef,
+): void => {
+  const options = useGridSelector(apiRef, optionsSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
+  const isInScrollBottomArea = React.useRef<boolean>(false);
+
+  const handleGridScroll = React.useCallback(
+    (event) => {
+      const scrollPositionBottom =
+        event.target.scrollTop + containerSizes?.windowSizes.height + options.scrollBottomThreshold;
+
+      if (containerSizes && scrollPositionBottom >= containerSizes.dataContainerSizes.height) {
+        if (!isInScrollBottomArea.current && options.onScrollBottom) {
+          isInScrollBottomArea.current = true;
+          options.onScrollBottom();
+        }
+      } else {
+        isInScrollBottomArea.current = false;
+      }
+    },
+    [options, containerSizes, isInScrollBottomArea],
```

Comment:
Refs don't need to be included in dependencies. The eslint rule shouldn't warn.

```suggestion
    [options, containerSizes],
```

Reply:
Done